### ISSUE TITLE
Use TLS Encryption to Protect RESTful API communications

### ISSUE THREAD

**Issue by zvigrinberg** [RHEcosystemAppEng/SaaSi#21]
needed to encrypt the communication as manifests  may contain sensitive data like secrets encoded in base64, and manifests will be sent in request payload to replica installer(especially important in case SaaSI engine and replica installer are not installed in the same cluster/same host).

**Comment by zvigrinberg** [comment]
In addition, passing sensitive parameters and their values  by users from client to service , should be also secured , by encapsulating them in HTTPS  TLS encrypted requests, otherwise payload data might be exposed to eavesdropping and interception.

**Comment by dmartinol** [comment]
I think we can integrate the install-builder as one component of the saasi-service, so it can just store the manifests in the management cluster, no need to have dedicated API to put these configurations in the engine.
And yes, the deployment configurations can be stored in a dedicated DB that can define the history of the custom replicas.

**Comment by zvigrinberg** [comment]
Right, but let's clarify few points tomorrow on our meeting, to get a clearer picture.

**Comment by zvigrinberg** [comment]
@dmartinol 
So for what concern sending the manifests with sensitive data, we'll do it  against api-server(port 6443) and this communication is already TLS encrypted by default.

But , We'll still need to protect The parameters submitting from SaaSI console to Exporter or one of its backend services endpoints (that will encrypt them and will create secrets from them eventually), the parameters will be sent over HTTP , and to protect them from interceptions and MITM Attacks, we need to encrypt that channel and send them over HTTPS using TLS.
